File: src/Zoo.java
import java.util.ArrayList;

public class Zoo {

    private ArrayList<Animal> animals;

    public static void main(String[] args){

        Zoo zoo = new Zoo();
        Zookeeper zookeeper = new Zookeeper(zoo);
        zookeeper.manageZoo();

    }

    Zoo(){
        this.animals = new ArrayList<>();

        // Pachyderms
        animals.add(new Elephant());
        animals.add(new Elephant());
        animals.add(new Hippo());
        animals.add(new Hippo());
        animals.add(new Rhino());
        animals.add(new Rhino());

        // Canines
        animals.add(new Dog());
        animals.add(new Dog());
        animals.add(new Wolf());
        animals.add(new Wolf());

        // Feline
        animals.add(new Cat());
        animals.add(new Cat());
        animals.add(new Lion());
        animals.add(new Lion());
        animals.add(new Tiger());
        animals.add(new Tiger());
    }

    /**
     * Get a list of all the animals in the zoo
     * @return animals (ArrayList<Animal>
     */
    public ArrayList<Animal> getAnimals() {
        return animals;
    }

    /**
     *  Set list of animals in the zoo to new list of animals
     * @param animals:ArrayList<Animal>
     */
    public void setAnimals(ArrayList<Animal> animals) {
        this.animals = animals;
    }

    /**
     *  Add one animal to the animals in the zoo
     * @param animal:Animal
     */
    public void addAnimal(Animal animal){
        this.animals.add(animal);
    }
}

File: Project2/Zoo.py
from Animal import Animal
from Canine import Canine
from Dog import Dog
from Wolf import Wolf
from Feline import Feline
from Cat import Cat
from Tiger import Tiger
from Lion import Lion
from Elephant import Elephant
from Hippo import Hippo
from Rhino import Rhino

class Zoo():

  '''
    Initializes Zoo by calling populate function 
  '''
  def __init__(self):
    self.animals = []
    self.populate()

  '''
    Adds an animal to the zoo
    @param animal:Animal
  '''
  def add_animal(self, animal):
    self.animals.append(animal) 

  '''
    Returns the animals in the zoo
    @return animals:list(Animal)  
  '''
  def get_animals(self):
    return self.animals

  '''
    Populates the zoo with two of each animal 
  '''
  def populate(self):

    # Canine
    self.animals.append(Dog())
    self.animals.append(Dog('Dug'))
    self.animals.append(Wolf())
    self.animals.append(Wolf('Wifi'))

    # Feline
    self.animals.append(Cat())
    self.animals.append(Cat('Catherine'))
    self.animals.append(Tiger())
    self.animals.append(Tiger('Terry'))
    self.animals.append(Lion())
    self.animals.append(Lion('Lenny'))
  
    # Pachyderm
    self.animals.append(Elephant())
    self.animals.append(Elephant('Esther'))
    self.animals.append(Hippo())
    self.animals.append(Hippo('Harriet'))
    self.animals.append(Rhino())
    self.animals.append(Rhino('Rose'))
    
  '''
    Deletes all animals in zoo and sets animals list to empty   
  '''
  def clear_animals(self):
    for animal in self.animals:
      del animal 
    self.animals = []

